librz/core/canalysis.c

```diff
@@ -7126,3 +7126,124 @@ RZ_API st64 rz_core_analysis_calls_count(RZ_NONNULL RzCore *core) {
 	}
 	return cov;
 }
+
+static const char *RzCoreAnalysisNameTypeStrs[] = {
+	"var",
+	"function",
+	"flag",
+	"address",
+};
+
+/**
+ * \brief Convert \p typ to string (const char*)
+ */
+RZ_API RZ_BORROW const char *rz_core_analysis_name_type_to_str(RzCoreAnalysisNameType typ) {
+	switch (typ) {
+	case RZ_CORE_ANALYSIS_NAME_TYPE_VAR:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_FUNCTION:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_FLAG:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_ADDRESS:
+		return RzCoreAnalysisNameTypeStrs[typ];
+	default:
+		rz_warn_if_reached();
+		return NULL;
+	}
+}
+
+RZ_API void rz_core_analysis_name_free(RZ_NULLABLE RzCoreAnalysisName *p) {
+	if (!p) {
+		return;
+	}
+	free(p->name);
+	free(p->realname);
+	free(p);
+}
+
+/**
+ * \brief Rename whatever var/flag/function is used at \p addr to \p name
+ * \return success?
+ */
+RZ_API bool rz_core_analysis_rename(RZ_NONNULL RzCore *core, RZ_NONNULL const char *name, ut64 addr) {
+	rz_return_val_if_fail(core && core->analysis && RZ_STR_ISNOTEMPTY(name), false);
+
+	ut8 buf[32];
```

Comment:
maybe we should use `rz_analysis_archinfo` + `RZ_ANALYSIS_ARCHINFO_MAX_OP_SIZE` to determine the maxsize of the buffer here? I'm not sure how it is done around the codebase.. I bet it's quite inconsistent. @wargio do you know?

Reply:
yes technically you should do that, but normally 128 bytes are always enough, 32 not so sure...

Reply:
@imbillow please use 128 for now... I guess we should at least define a macro in rz_analysis.h in the future for that magic number.

Reply:
OK